### ISSUE TITLE
Live shared links?

### ISSUE THREAD

**Issue by travisdowns** [mattgodbolt/gcc-explorer#173]
Question: is it possible to share a "live" link to godbolt? That is, a link that will reflect modification if the code is modified.

As far as I understand, all shared links are currently snapshots of the code in the RHS pane and the compiler and options selections.

**Comment by mattgodbolt** [comment]
There isn't as yet: no. I deliberately haven't relied on shared storage on the CE servers yet but I'm starting to explore that idea a little; which would allow this.

**Comment by travisdowns** [comment]
Good point - I guess the existing `godbolt.org` shortlinks simply encode the entire state of the linked workspace into the URL itself, so no actual local storage is needed. Well, I guess consider this is a feature-request or just close it or whatever. Either way, godbolt is aweome!

**Comment by mattgodbolt** [comment]
An update on this: CE now handles its own links internally and we have had (some) hacky solutions for "here's a named link, and I can change its contents later". It's all very hacky and requires manual intervention on our side, but as part of future changes I'd like to better support this. @travisdowns ; if you have a couple of these you'd like to try out we can give it a go (e.g. godbolt.org/z/travisdowns/blah) ?

**Comment by travisdowns** [comment]
@mattgodbolt - I don't have a "real" one handy this moment, but we can use any old thing if the purpose is to test it out, right? I'm sure I'll have a real one soon enough.

We could use this for testing:

https://godbolt.org/z/kVCPYD

**Comment by mattgodbolt** [comment]
Zoiks I can't believe I didn't get back to you on this. Sorry @travisdowns ... And we're still not near releasing this more generally. We need a login and self-service thing I think to support this...one day!

**Comment by travisdowns** [comment]
> We need a login and self-service thing I think to support this...one day!

Right, I guess the current "read only" approach neatly gets around the question of who can modify what, logins, etc.

Just ping me if you ever want some help testing anything out!

**Comment by RubenRBS** [comment]
No idea if it would even begin to fit our usecase, but I've found https://github.com/convergencelabs/monaco-collab-ext which went opensource some time ago

**Comment by matsjla** [comment]
There's also the integration powering VSCode Live Share that might be of interest https://github.com/MicrosoftDocs/live-share
